src/ceph-volume/ceph_volume/util/arg_validators.py

```diff
@@ -51,13 +52,48 @@ def _is_valid_device(self):
         return self._device
 
 
+def get_osd_ids_up():
+    bootstrap_keyring = '/var/lib/ceph/bootstrap-osd/%s.keyring' % conf.cluster
+    cmd = ['ceph', '--cluster', conf.cluster,
+            '--name', 'client.bootstrap-osd',
+            '--keyring', bootstrap_keyring,
+            'osd', 'tree', '-f', 'json']
+    stdout, stderr, returncode = process.call(
+        cmd,
+        show_command=False
+    )
+
+    if returncode != 0:
+        raise RuntimeError(f"Couldn't run command: '{' '.join(cmd)}'")
+
+    output = json.loads(''.join(stdout).strip())
+    osds = output['nodes']
+    return [osd["id"] for osd in osds if osd['type'] == "osd"]
+    return output
```

Comment:
unreachable return statement?

Reply:
yeh, looks like a leftover from when I was trying the first implementation of this.